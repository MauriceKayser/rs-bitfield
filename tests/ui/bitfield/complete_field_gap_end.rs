//! Assertion failure: _COMPLETE_FIELD_0_MUST_NOT_HAVE_GAPS

extern crate alloc;

#[bitfield::bitfield(8)]
struct BitField(#[field(size = 2, complete)] Field); // Field value `3` is missing.

#[derive(Clone, Copy, bitfield::Field)]
#[repr(u8)]
enum Field {
    F0,
    F1,
    F2
}

fn main() {}