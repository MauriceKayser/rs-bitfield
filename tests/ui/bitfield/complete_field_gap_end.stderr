error[E0080]: evaluation of constant value failed
 --> tests/ui/bitfield/complete_field_gap_end.rs:6:35
  |
6 | struct BitField(#[field(size = 2, complete)] Field); // Field value `3` is missing.
  |                                   ^^^^^^^^ the evaluated program panicked at 'Complete field must not have gaps', $DIR/tests/ui/bitfield/complete_field_gap_end.rs:6:35
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)
